moveit_simple_msgs/srv/TrajectoryTrainer.srv

```diff
@@ -0,0 +1,19 @@
+uint8 NEW_WP=0				# Create a new waypoint at current location 			[required: waypoint name, optional: trajectory name]
+uint8 DEL_WP=1				# Delete existing waypoint(s) from config 				[required: waypoint name(s)]
+uint8 GET_WP=2				# Request information on existing waypoint(s) 			[required: waypoint name(s)]
+uint8 UPDATE_WP=3			# Replace previous waypoint data with current location 	[required: waypoint name]	
+uint8 APPEND_WP_TO_TRAJ=4	# Add waypoint(s) to trajectory                         [required: trajectory name, waypoint name(s)] 
+uint8 REMOVE_WP_FROM_TRAJ=5 # Remove waypoint(s) from trajectory if it(they) exist 	[required: trajectory name, waypoint name(s)]
+uint8 NEW_TRAJ=10			# Create a new trajectory, optionally populate          [required: trajectory name, optional: waypoint name(s)]
+uint8 DEL_TRAJ=11			# Delete a trajectory from config						[required: trajectory name]
+uint8 GET_TRAJ=12			# Get all waypoints in a trajectory 					[required: trajectory name]
+uint8 CLEAR_TRAJ=13			# Clears the contents of a trajectory 					[required: trajectory name]
+uint8 RELOAD_YAML=255		# Update internal storage to match YAML                 [required: N/A]
+
+uint8 op_type
+string trajectory_name
```

Comment:
Under what circumstances are these member variables filled out?  Should we consider creating multiple services?

Reply:
The formatting is off, but the comments specify what elements are necessary during the call. When performing trajectory manipulations, you need the trajectory name and sometimes waypoint names. Similar for waypoint manipulations. op_type is always required.

Reply:
As an outsider observer: these sort of "union services" seem to go against the idea of having simple, explicit and clearly defined semantics. The interpretation of the fields in the request change (dramatically almost) based on the value of `op_type`. The type of the service itself does not carry enough semantic meaning to sufficiently explain the cases in which this can be used, or what to expect from an invocation of the service.

It might be worthwhile to provide additional services that implement some of this functionality and give them types that unambiguously convey their semantics. That is both user friendly *and* reduces the opportunity for mistakes when filling out fields (and the need the `op_type` field).

Finally: this might actually be a use-case for Actions, as some of these operations can potentially take a long time (fi: depending on how large the `.yaml` file is that is being loaded (long traj?) it could take going to disk, loading it, parsing, etc. All this time the caller is blocked.

Reply:
Alright, alright, I may have shoe-horned it a little bit. From the current MoveIt Simple perspective, I see get_waypoint (request a single waypoint) as critical for the lookupTrajectoryPoint extension. Additionally, I think get_waypoints (request multiple, possibly unrelated waypoints), and get_trajectory (request multiple, related waypoints) would be useful conveniences for building a trajectory.

As for using an action server, the extension to lookupTrajectoryPoint should block until a response regardless, no?